src/lib/mina_base/parties_logic.ml

```diff
@@ -817,6 +838,48 @@ module Make (Inputs : Inputs_intf) = struct
       let a = Account.set_timing timing a in
       (a, local_state)
     in
+    (* Transform into a snapp account.
+       This must be done before updating snapp fields!
+    *)
+    let a = Account.make_snapp a in
```

Comment:
Can we do this only if there's an update? or prioritize the change to allow payments to/from snapp accounts?  The way it is now makes accounts, if updated in parties transaction, unusable for normal payments. For example, fee payer accounts that most probably won't be snapp accounts or payment receivers that are part of a snapp transaction

Reply:
nvm, `unmake_snapp` added [here](#10218) works.